lib/postprocessing/image.js

```diff
@@ -142,9 +191,45 @@ function hasJpegQuality(metadata, instructions) {
     return true;
 }
 
+/**
+ * Returns updated rendition instructions for the worker callback processing if needed
+ * or undefined if no changes are required.
+ */
+async function prepareImagePostProcess(rendition, options) {
+    if (rendition.instructions.fmt && isSupportedOutputFormat(rendition.instructions)) {
+
+        const supportedFormats =
+            options.supportedRenditionFormats
+            || WORKER_SUPPORTED_FMT[new OpenwhiskActionName().name]
+            || DEFAULT_WORKER_SUPPORTED_FMT;
+
+        // in case the worker cannot generate the final rendition format itself, we have to do it via post processing
+        if (!supportedFormats.includes(rendition.instructions.fmt)) {
+            // ...and tell the worker to generate a different intermediate format it can do
+            // find first format that is supported (worker defines priority order)
+            const newFormat = supportedFormats.find(fmt => SUPPORTED_INPUT_FMT.includes(fmt));
+            if (!newFormat) {
+                throw new GenericError(`Worker does not support generating a compatible image format. Requested: ${SUPPORTED_INPUT_FMT} Supported by worker: ${supportedFormats}`);
```

Comment:
Since it's an array it ends up being something like this:

```
Worker does not support generating a compatible image format. Requested: ["png", "jpg"] Supported by worker: ["tif"]
```

so I thought this has a visual delimiter already. No strong opinion.